proto/minder/v1/minder.proto

```diff
@@ -1784,7 +1823,72 @@ message ListProvidersResponse {
     string cursor = 2;
 }
 
-// ProviderType is the type of the provider.
+message CreateProviderRequest {
+    // context is the context in which the provider is created.
+    Context context = 1;
+    // provider is the provider to be created.
+    Provider provider = 2;
+}
+
+message CreateProviderResponse {
+    // provider is the provider that was created.
+    Provider provider = 1;
+
+    // authorization provides additional authorization information needed
+    // to complete the initialization of the provider.
+    AuthorizationParams authorization = 2;
+}
+
+message DeleteProviderRequest {
+    // context is the context in which the provider is deleted.  Both
+    // project and provider are required in this context.
+    Context context = 1;
+}
+
+message DeleteProviderResponse {
+}
+
+message GetUnclaimedProvidersRequest {
+    // context is the context in which the set of providers are evaluated.
+    Context context = 1;
+}
+
+message GetUnclaimedProvidersResponse {
+    // providers is a set of parameters which can be supplied to allow the user
+    // to assign existing unclaimed credentials to a new provider in the project
+    // via CreateProvider().
+    repeated ProviderParameter providers = 1;
+}
+
+message AuthorizationParams {
+    // authorization_url is an external URL to use to authorize the provider.
+    string authorization_url = 1;
+
+}
+
+message ProviderParameter {
+    oneof parameters {
+        GitHubAppParams github_app = 1;
+    }
+}
+
+// GitHubAppParams is the parameters for a GitHub App provider.
+message GitHubAppParams {
+    // The GitHub installation ID for the app.  On create, this is the only
+    // parameter used; the organization parameters are ignored.
+    int64 installation_id = 1;
+
+    // The GitHub organization slug where the app is installed.  This is an
+    // output-only parameter, and is validated on input if set (i.e. the value
+    // must be either empty or match the org of the installation_id).
+    string organization = 2;
+    // The GitHub organization ID where the app is installed.  This is an
+    // output-only parameter, and is validated on input if set (i.e. the value
+    // must be either empty or match the org of the installation_id).
+    int64 organization_id = 3;
```

Comment:
If an organization is renamed, the `organization_id` will remain consistent, while the `organization` string in the output may change.

Thinking about it further, as I implement this, I may change the rule from "output only" to "must all agree" -- i.e. if the user sets `installation` and `organization_id`, they both need to match up.  That would allow sophisticated clients to protect against repo rename attacks, while simple clients can just supply the parameter(s) which are easiest.

Reply:
I'm wondering if we need both `installation_id` and `organization_id` since there a one to one mapping between them.
And I assume we only want to display `organization` to the end user?

Reply:
I wasn't clear if we'd want to return both for the CLI.  Given that we're already collecting that data, it seemed cheap to present it.